flink-end-to-end-tests/test-scripts/common_kubernetes.sh

```diff
@@ -39,8 +39,12 @@ function setup_kubernetes_for_linux {
     if ! [ -x "$(command -v kubectl)" ]; then
         echo "Installing kubectl ..."
         local version=$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)
-        curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/$version/bin/linux/$arch/kubectl && \
-            chmod +x kubectl && sudo mv kubectl /usr/local/bin/
+        download_kubectl_url="https://storage.googleapis.com/kubernetes-release/release/$version/bin/linux/$arch/kubectl"
+        if ! retry_times 3 5 "curl -Lo kubectl ${download_kubectl_url}"; then
```

Comment:
```suggestion
        if ! retry_times 3 5 "curl --fail -Lo kubectl ${download_kubectl_url}"; then
```
Looks like this PR has the same issue like PR #23497